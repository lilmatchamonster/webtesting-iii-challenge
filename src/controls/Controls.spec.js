// Test away!
import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';
import 'jest-dom/extend-expect';
import renderer from 'react-test-renderer'
import Controls from '../controls/Controls'

describe('Testing Controls Module', () => {

  it('should render component on load', () => {
    render(<Controls />);
  });

  it('should display lock gate if locked is false and closed is true', () => {
    const locked = false;
    const closed = true;

    const { getByText } = render(<Controls locked={locked} closed={closed} />);

    const text = getByText(/lock gate/i);

    expect(text).toBeInTheDocument();
  })

  it('should display open gate if locked is false and closed is true', () => {
    const locked = false;
    const closed = true;

    const { getByText } = render(<Controls locked={locked} closed={closed} />);

    const text = getByText(/open gate/i);

    expect(text).toBeInTheDocument();
  })

  it('should match the snapshot', () => {
    const tree = renderer.create(<Controls />);
    expect(tree.toJSON()).toMatchSnapshot();
  })

});