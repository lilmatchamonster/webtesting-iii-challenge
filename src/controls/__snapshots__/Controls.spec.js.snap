// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Controls Module should match the snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
  >
    Close Gate
  </button>
</div>
`;
